trestle/core/commands/author/component.py

```diff
@@ -0,0 +1,290 @@
+from trestle.core.commands.author.common import AuthorCommonCommand
+from trestle.core.commands.common.return_codes import CmdReturnCodes
+from trestle.core.control_interface import ContextPurpose, ControlContext, ControlInterface
+from trestle.core.models.file_content_type import FileContentType
+from trestle.core.profile_resolver import ProfileResolver
+from trestle.oscal import OSCAL_VERSION
+
+logger = logging.getLogger(__name__)
+
+
+class ComponentGenerate(AuthorCommonCommand):
+    """Generate component in markdown form from a component in the trestle workspace."""
+
+    name = 'component-generate'
+
+    def _init_arguments(self) -> None:
+        name_help_str = 'Name of the source component model in the trestle workspace'
+        self.add_argument('-n', '--name', help=name_help_str, required=True, type=str)
+        profile_help_str = 'Name of the profile model in the trestle workspace'
+        self.add_argument('-p', '--profile', help=profile_help_str, required=True, type=str)
+        self.add_argument('-o', '--output', help=const.HELP_MARKDOWN_NAME, required=True, type=str)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        try:
+            log.set_log_level_from_args(args)
+            trestle_root = args.trestle_root
+            markdown_dir_name = args.output
+            profile_name = args.profile
+            component_name = args.name
+
+            return self.component_generate_all(
+                trestle_root, component_name, profile_name, markdown_dir_name, {}, None, False
+            )
+
+        except Exception as e:  # pragma: no cover
+            return handle_generic_command_exception(e, logger, 'Generation of the component markdown failed')
+
+    def component_generate_all(
+        self,
+        trestle_root: pathlib.Path,
+        comp_def_name: str,
+        profile_name: str,
+        markdown_dir_name: str,
+        yaml_header: Dict[str, Any],
+        sections_dict: Optional[Dict[str, str]],
+        overwrite_header_values: bool
+    ) -> int:
+        """Generate markdown for all components in comp def."""
+        if not file_utils.is_directory_name_allowed(markdown_dir_name):
+            raise TrestleError(f'{markdown_dir_name} is not an allowed directory name')
+        md_path = trestle_root / markdown_dir_name
+        md_path.mkdir(parents=True, exist_ok=True)
+        component_def, _ = load_validate_model_name(trestle_root, comp_def_name, comp.ComponentDefinition)
+        for component in as_list(component_def.components):
+            rc = self.component_generate_by_name(
+                trestle_root,
+                comp_def_name,
+                component.title,
+                profile_name,
+                md_path / component.title,
+                yaml_header,
+                sections_dict,
+                overwrite_header_values
+            )
+            if rc != CmdReturnCodes.SUCCESS.value:
+                return rc
```

Comment:
Please simplify to one return at the very end
```
return_code = CmdReturnCodes.SUCCESS.value
..
..
return_code= self.component_generate_by_name()
if return_code != CmdReturnCodes.SUCCESS.value:
                  break
...
return return_code
```

Reply:
fixed